### PR TITLE
chore: upgrade snakenet to polkadot v0.9.15

### LINKED ISSUE #357: upgrade snakenet solochain to polkadot 0.9.15
storage should be as close as possible to parachain for the following data migration

### PR DESCRIPTION
**Pull request by enthusiastmartin**
fixes #357

### CODE REVIEW

**Comment by enthusiastmartin** [comment]
@martinfridrich if you can have a look at the chain spec, that would be great.

File: node/src/command.rs [hunk 1]
```diff
@@ -92,24 +92,19 @@ impl SubstrateCli for Cli {
...
-
 /// Parse and run command line arguments
 pub fn run() -> sc_cli::Result<()> {
 	let cli = Cli::from_args();
-	set_ss58();
 
 	match &cli.subcommand {
 		None => {
-			let run_testing_runtime = cli.run.runtime.is_testing_runtime();
+			//let runner = cli.create_runner(&cli.run.base.normalize())?;
```
Comment (lumir-mrkva) [inline node/src/command.rs:101]: remove

File: node/src/command.rs [hunk 2]
```diff
@@ -92,24 +92,19 @@ impl SubstrateCli for Cli {
...
 			let runner = cli.create_runner(&cli.run.base)?;
+
 			runner.run_node_until_exit(|config| async move {
 				match config.role {
-					Role::Light => service::build_light(config, run_testing_runtime),
-					_ => service::build_full(config, run_testing_runtime).map(|full| full.task_manager),
+					Role::Light => panic!("Light is not supported"),
+					_ => service::build_full(config, false).map(|full| full.task_manager),
```
Comment (lumir-mrkva) [inline node/src/command.rs:107]: now, how do you run testing runtime? :)

File: primitives/src/constants.rs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	pub const MAX_IN_RATIO: u128 = 3;
+
+	/// Trading limit
+	pub const MIN_TRADING_LIMIT: Balance = 1000;
+
+	/// Minimum pool liquidity
+	pub const MIN_POOL_LIQUIDITY: Balance = 1000;
+
+	/// We allow for
+	pub const MAXIMUM_BLOCK_WEIGHT: Weight = WEIGHT_PER_SECOND / 2;
```
Comment (lumir-mrkva) [inline primitives/src/constants.rs:38]: 0.5 sec is limitation of parachain, solochain is fine with 2 sec processing time

```suggestion
	pub const MAXIMUM_BLOCK_WEIGHT: Weight = WEIGHT_PER_SECOND * 2;
```

File: pallets/claims/src/lib.rs [hunk 4]
```diff
@@ -231,7 +233,7 @@ where
 		_len: usize,
 	) -> TransactionValidity {
 		match call.is_sub_type() {
-			Some(Call::claim(signature)) => match Pallet::<T>::validate_claim(who, signature) {
+			Some(Call::claim{ethereum_signature}) => match Pallet::<T>::validate_claim(who, ethereum_signature) {
```
Comment (lumir-mrkva) [inline pallets/claims/src/lib.rs:236]: why was this renamed? isn't it breaking any API? https://claim.hydradx.io

File: runtime/common/src/lib.rs [hunk 5]
```diff
@@ -282,9 +284,78 @@ parameter_types! {
...
+	}
+}
+
+
+use pallet_staking::BenchmarkingConfig;
+
+pub struct StakingBenchmarkingConfig;
+
+impl BenchmarkingConfig for StakingBenchmarkingConfig{
+	type MaxValidators = ConstU32<100>;
```
Comment (lumir-mrkva) [inline runtime/common/src/lib.rs:342]: is this accurate for our case? we should have like 165 validators with 64 nominators each. anyway it requires rebench probably

File: runtime/hydra-dx/src/lib.rs [hunk 6]
```diff
@@ -145,8 +160,8 @@ pub fn native_version() -> NativeVersion {
 
 pub struct BaseFilter;
 
-impl Filter<Call> for BaseFilter {
-	fn filter(call: &Call) -> bool {
+impl Contains<Call> for BaseFilter {
+	fn contains(call: &Call) -> bool {
 		match call {
 			Call::Authorship(_)
```
Comment (lumir-mrkva) [inline runtime/hydra-dx/src/lib.rs:166]: note that rest of the calls dont have to be listed in match and you can just match rest (_) as true

File: runtime/hydra-dx/src/lib.rs [hunk 7]
```diff
@@ -545,29 +596,34 @@ parameter_types! {
...
 	type SolutionImprovementThreshold = MinSolutionScoreBump;
-	type MinerMaxIterations = MinerMaxIterations;
 	type MinerMaxWeight = MinerMaxWeight;
 	type MinerTxPriority = MultiPhaseUnsignedPriority;
 	type DataProvider = Staking;
-	type OnChainAccuracy = Perbill;
-	type CompactSolution = NposCompactSolution16;
-	type Fallback = Fallback;
-	type BenchmarkingConfig = ();
+	type Fallback = pallet_election_provider_multi_phase::NoFallback<Self>;
```
Comment (lumir-mrkva) [inline runtime/hydra-dx/src/lib.rs:625]: fallback should be same as before (on-chain)

File: pallets/claims/src/migration.rs [hunk 8]
```diff
@@ -16,13 +16,17 @@
...
 
 pub fn import_initial_claims<T: Config>(claims_data: &[(&'static str, Balance)]) -> frame_support::weights::Weight {
-	let version = <Pallet<T> as GetPalletVersion>::storage_version();
-	if version == None {
+	let version = <Pallet<T> as GetStorageVersion>::current_storage_version();
+
+	//TODO: it used to return None if it did not exists, so what do now ?
+	// I believe
+
+	if version == 0 {
```
Comment (lumir-mrkva) [inline pallets/claims/src/migration.rs:29]: check storage in mainnet, make sure that we don't run this migration again. theoretically, this migration could be removed already, with all data related

File: node/src/service.rs [hunk 9]
```diff
@@ -124,14 +132,15 @@ pub fn new_partial<RuntimeApi, Executor>(
...
+			Option<TelemetryWorkerHandle>,
 		),
 	>,
 	ServiceError,
 >
 where
 	RuntimeApi: ConstructRuntimeApi<Block, FullClient<RuntimeApi, Executor>> + Send + Sync + 'static,
 	RuntimeApi::RuntimeApi: RuntimeApiCollection<StateBackend = sc_client_api::StateBackendFor<FullBackend, Block>>,
-	Executor: NativeExecutionDispatch + 'static,
+	Executor: NativeExecutionDispatch + 'static, //+ sp_core::traits::CodeExecutor + sc_executor::RuntimeVersionOf ,
```
Comment (lumir-mrkva) [inline node/src/service.rs:143]: remove comment

File: pallets/claims/src/tests.rs [hunk 10]
```diff
@@ -112,11 +113,12 @@ fn signed_extention_success() {
 	new_test_ext().execute_with(|| {
 		let signature = hex!["5b2b46b0162f4b4431f154c4b9fc5ba923690b98b0c2063720799da54cb35a354304102ede62977ba556f0b03e67710522d4b7523547c62fcdc5acea59c99aa41b"];
 
-		let call = <crate::Call<Test>>::claim(EcdsaSignature(signature)).into();
+		//let call = <pallet_claims::Call<Test>>::claim(EcdsaSignature(signature));//.into();
```
Comment (lumir-mrkva) [inline pallets/claims/src/tests.rs:116]: remove

File: runtime/hydra-dx/src/tests.rs [hunk 11]
```diff
@@ -15,9 +14,9 @@ fn full_block_cost() {
 
 	let max_weight = BlockWeights::get().get(DispatchClass::Normal).max_total.unwrap_or(1);
 	let weight_fee = WeightToFee::calc(&max_weight);
-	assert_eq!(weight_fee, 19149775500000);
+	assert_eq!(weight_fee, 120000000000000);
```
Comment (lumir-mrkva) [inline runtime/hydra-dx/src/tests.rs:17]: does this mean that we're basically increasing-price 10 fold?

**Review by lumir-mrkva — CHANGES_REQUESTED** [review]
- [ ] bump versions
- [ ] reuse or remove asset registry, xyk and exchange

File: pallets/claims/src/lib.rs (hunk 4 above)
Comment (enthusiastmartin) [reply pallets/claims/src/lib.rs:236]: It was not renamed . it is actually called ethereum_signature in the claim extrinsic.

the reason it had to be renamed in the match statement  is that it had to be changed to use  struct pattern syntax. 

so instead of using this:
```rust
Some(Call::claim{ethereum_signature: signature}) => match Pallet::<T>::validate_claim(who, signature)
```

i went with just eithereium signature.

File: runtime/hydra-dx/src/lib.rs (hunk 7 above)
Comment (enthusiastmartin) [reply runtime/hydra-dx/src/lib.rs:625]: Well, Fallback strategy where it originally took the OnChain fallback has been removed. 

any tip how to do ?

File: runtime/hydra-dx/src/tests.rs (hunk 11 above)
Comment (enthusiastmartin) [reply runtime/hydra-dx/src/tests.rs:17]: the difference is because of removal of extra weight from ExtrinsicBaseWeight.

If i  add back the extra weight - it is the same result 19149775500000.

Which is kind of surprising 😕

File: runtime/common/src/lib.rs (hunk 5 above)
Comment (enthusiastmartin) [reply runtime/common/src/lib.rs]: Set 165 and 64.

File: runtime/hydra-dx/src/lib.rs (hunk 7 above)
Comment (enthusiastmartin) [reply runtime/hydra-dx/src/lib.rs:602]: leaving to NoFallback.

File: pallets/claims/src/migration.rs (hunk 8 above)
Comment (enthusiastmartin) [reply pallets/claims/src/migration.rs:29]: removed on_runtime_upgrade hook.

File: node/src/command.rs (hunk 2 above)
Comment (Roznovjak) [reply node/src/command.rs]: done

**Review by lumir-mrkva — APPROVED** [review]
